freqtrade/enums/signaltype.py

```diff
@@ -5,12 +5,20 @@ class SignalType(Enum):
     """
     Enum to distinguish between enter and exit signals
     """
-    BUY = "buy"
-    SELL = "sell"
+    ENTER_LONG = "enter_long"
+    EXIT_LONG = "exit_long"
+    ENTER_SHORT = "enter_short"
+    EXIT_SHORT = "exit_short"
 
 
 class SignalTagType(Enum):
     """
     Enum for signal columns
     """
-    BUY_TAG = "buy_tag"
+    LONG_TAG = "long_tag"
+    SHORT_TAG = "short_tag"
```

Comment:
@samgermain Maybe a strange question at this point...
do we really need 2 seperate columns for this?

if long and short entries collide, we'll reject the trade anyway (it's a shitty strategy which isn't sure wether to go long or short), so maybe having this column named `entry_tag` instead would simplify usage from the strategy a bit (one fewer column-name to remember for users)...

Reply:
Hello @xmatthias take a look at [this]( https://youtu.be/Yq-m6IQ3FHw).
As you can see having both position open for a period can be a legit strategy 🙂

Reply:
> Hello @xmatthias take a look at [this](https://youtu.be/Yq-m6IQ3FHw).
> As you can see having both position open for a period can be a legit strategy 🙂

We're not dealing with hedge mode for now, to keep the complexity down

Reply:
@Axel-CH if both `enter_long` and `enter_short` values are 1 for the same candle, the bot will do nothing, as it's a conflicting signal.

Hedge-mode is not in scope for the moment (and probably won't be for a long time) - so it'll be one position (side) per trade only - in which case having split columns will not give us many benefits.

Reply:
@xmatthias I totally understand the necessity to keep things simple. As you are maintaining this project for years now, you have a good view of what can add complexity and complicate debugging.

However, just on the strategy point of view I'm not shocked by something like: "I want to enter_long at this price and short at that price" In that case both order will be sent to the exchange but it's the market price that will decide which one is filled.

Anyway, as you said it's a specific case and if it's adding to much complexity, it's a good question to ask if it worth it.

Reply:
this case is a different one alltogether - you'd place two orders (one long, one short) - and assume that the filled one will cancel the other one (?)

I'm not sure that'll be a good approach ... as that assumes that price is moving against you - so you're long when price falls, and short when price raises (that's the wrong way around - at least when trying to make profits!)

A "real" Hedge would have you in both positions at the same time (assume both orders fill) - but unless explicitly allowed by the exchange (binnace hedge mode), that's not something that'll work as the 2nd fill will close the other position (you buy 100 XRP, and sell 100XRP - so your position is now 0 XRP).

Reply:
When the price moove in trend, there is two types of price movements.
Impulsion: The price strongly follow trend direction 

Correction: The price perform a pullback as traders take profit at key levels 

If you are greedy (😂)you want to take advantage of both cases. 

The impulsion wave is where you want to make your profit. But you can use the correction to make a better entry and ride the next impulsion wave. So yes you are entering in the trade when the price go against you for a short time, but you still follow the overall trend. 

In that case if you also want to take advantage of the correction wave you can enter position at the begining of it and exit at the same level you want to enter the trend following order. So technically freqtrade will have both "position" open. But in reality only one side is filled and active on exchange. 

I Will do a shema to make it clearer

Reply:
Also, to prevent unwanted opening of a position on the opposition side, we can use the "reduce only" option when we create orders. [More details ](https://help.bybit.com/hc/en-us/articles/360039260574-What-is-a-Reduce-Only-Order-)
I think we should add it by default

Reply:
Freqtrade does not support bybit at this point - so i don't see this order-type as relevant.

even in your case (which for sure will not be supported to begin with) - it'd be the same signal that indicates buying and selling (indicates this type of entry) - so i don't really see the need to have 2 seperate columns for the entry reason.

It's a case / scenario where i feel we're trying to prepare for an eventual future which potentially never arrives - and as a consequence, making our life (or the code) more complex than how it really should be.

Reply:
Reduce only orders are available on binance, ftx, kraken...etc

Reply:
that's totally unrelated to the discussion here ... so i don't see how this can help in deciding if short_tag needs to be a seperate column or not.

Reply:
I just checked on the create_trade code. To manage my specific case of having two signals on the same candle an option could be added to create _trade to specify the side wanted.

In that case global tag for long and short won't be an issue.
The downside will be the duplicate execution of all code inside create_trade function.

But as we agreed to not do
(enter_long, enter_short, exit_long, exit_short) = get_signal()

Having only one entry_tag seem logic to me

Reply:
I spoke about the "reduceOnly" option to answer to your question as this feature prevent orders to conflict

"this case is a different one alltogether - you'd place two orders (one long, one short) - and assume that the filled one will cancel the other one (?)"
👍🏽